### ISSUE TITLE
[Question] Why do you need connectionUrl from source?

### ISSUE THREAD

**Issue by dmvieira** [MarquezProject/marquez#864]
Isn't it a security breach? Why exactly Marquez need to access source?

**Comment by julienledem** [comment]
Hi @dmvieira,
Marquez does not need access to source. `connectionUrl` is there for reference as an identifier of the datasource where the dataset is coming from. This URL is not supposed to contain credentials.

**Comment by wslulciuc** [comment]
> This URL is not supposed to contain credentials.

@julienledem: Agreed, though it can be a security concern as the username / password might be present in the connection URL. And, currently, the backend doesn't strip or redact the credentials if they're present.

Should we open an issue to strip out credentials in the connection if present?

**Comment by dmvieira** [comment]
No? I was just reading API documentation
![Screenshot_20201207-215031-046](https://user-images.githubusercontent.com/1108613/101422811-5da23380-38d6-11eb-9857-cbbf8c36d730.png)

Why not change this name to "sourceIdentifier" or something like that?

**Comment by wslulciuc** [comment]
Yeah, I think we can word this a bit better. The URL can, _technically_, still be used to connect to the source, but the credentials wouldn't be stored in Marquez. For example, if I wanted to verify that an upstream job dependency I have successfully inserted all rows to a table, I can get the connection URL from Marquez, then verify the row count myself (assuming I had the credentials handy). Same idea can be applied to jobs. That is, a job can call out to Marquez programmatically to get the source info for a dataset my job needs to read as input, while assuming the credentials are present on the server running my job.

**Comment by wslulciuc** [comment]
> Why not change this name to "sourceIdentifier" or something like that?

The source name is how Marquez uniquely identifies the source, but maybe just renaming `connectionUrl` to `url` would clarify it's usage?

**Comment by dmvieira** [comment]
I really don't understand why only source name is not enough... If I know my service I know it credentials and connection details.

I was looking other issues like https://github.com/MarquezProject/marquez/issues/698 and I was thinking how `connectionUrl` is applicable to a Filesystem, Cassandra, Hadoop Spark job or other distributed resources.

If I really understand, in fact do you want to map metadata describing de source and not act like a credentials storage. What do you think about enable more flexible metadata instead of turn `connectionUrl` mandatory?
